io/disk/parallel_dd.py

```diff
@@ -0,0 +1,231 @@
+                cmd += " %s=%s" % (option.split(":")[0],
+                                   option.split(":")[1])
+            # Wait for everyone to complete
+            proc = process.get_sub_process_klass(cmd)(cmd + ' > /dev/null',
+                                                      shell=True)
+            proc.start()
+            proc.poll()
+            proc.wait()
+        sys.stdout.flush()
+        sys.stderr.flush()
+
+    def fs_read(self):
+        """
+        Read in 'streams' files in parallel background tasks.
+        """
+        for i in range(self.streams):
+            s_file = os.path.join(self.srcdir, 'poo%d' % (i + 1))
+            cmd = 'dd if=%s of=/dev/null bs=4k count=%d' % \
+                (s_file, self.blocks_per_file)
+            for option in self.fs_dd_roptions.split():
+                cmd += " %s=%s" % (option.split(":")[0],
+                                   option.split(":")[1])
+            if self.seq_read:
+                process.run(cmd + ' > /dev/null', shell=True)
+            else:
+                # Wait for everyone to complete
+                proc = process.get_sub_process_klass(cmd)(cmd + ' > /dev/null',
+                                                          shell=True)
+                proc.start()
+                proc.poll()
+                proc.wait()
+            sys.stdout.flush()
+
+    def _device_to_fstype(self, s_file, device=None):
+        """
+        Checks for the filesystem and returns the type of the filesystem.
+        """
+        if not device:
+            device = self.fsys.device
+        try:
+            line = process.system_output('egrep ^%s %s' % (device, s_file))
+            logging.debug(line)
+            fstype = line.split()[2]
+            logging.debug('Found %s is type %s from %s', device, fstype,
+                          s_file)
+            return fstype
+        except process.CmdError:
+            logging.error('No %s found in %s', device, s_file)
+            return None
+
+    def test(self):
+        """
+        Test Execution.
+        """
+        try:
+            self.fsys.unmount()
+        except process.CmdError:
+            pass
+
+        self.log.info('------------- Timing raw operations ------------------')
+        start = time.time()
+        self.raw_write()
+        self.raw_write_rate = self.megabytes / (time.time() - start)
+
+        start = time.time()
+        self.raw_read()
+        self.raw_read_rate = self.megabytes / (time.time() - start)
+
+        self.fsys.mkfs(self.fstype)
+        self.fsys.mount(None)
+
+        self.log.info('------------- Timing fs operations ------------------')
+        start = time.time()
+        self.fs_write()
+        self.fs_write_rate = self.megabytes / (time.time() - start)
+        self.fsys.unmount()
+
+        self.fsys.mount(None)
+        start = time.time()
+        self.fs_read()
+        self.fs_read_rate = self.megabytes / (time.time() - start)
+
```

Comment:
Since you are doing the same thing multiple times here, can you optimise this code ?